### ISSUE TITLE
[MU4 Issue] Slurs/Ties should be on top layer comparing with staff lines (not behind staff when selected) 

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#11964]
![slur](https://user-images.githubusercontent.com/90187801/172891382-b6acbbf4-27c0-4169-b620-8a89c4fb0b69.PNG)

**Platform information**
macOS, Windows, Linux

**Additional context**
The same in **MU3**

**Comment by oktophonie** [comment]
This is totally fine as it is.

**Comment by DmitryArefiev** [comment]
Yeah) I'll edit summary to to make it clearer

**Comment by Jojo-Schmitz** [comment]
This about the z-order of slurs, right? Changing that from 1000 to 1300 (or more, but not less) should fix this issue.
Ties have the same issue, here an increase from 1100 to 1300 )again: or more, nit less) should fix it.

**Comment by DmitryArefiev** [comment]
> This about the z-order of slurs, right? Changing that from 1000 to 1300 (or more, but not less) should fix this issue. Ties have the same issue, here an increase from 1100 to 1300 (again: or more, not less) should fix it.

Yes. Thanks for ties, will add it

**Comment by MarcSabatella** [comment]
Please do *not* change this - it's quite deliberate and important that it it the way it is.  The idea is that we need to support the ability to do this:

![image](https://user-images.githubusercontent.com/1799936/173091614-7148871a-6b37-41c1-80f7-89a153fd633d.png)

I'm talking about how the ties "break" for the notes.  Similarly for ties that need to break across time signature or key signature changes.  Currently, way we support this is by placing an opaque white graphic at the correct Z-order such that it is behind the staff lines and everything else but in front of the slurs and ties.

Eventually of course it would be great to have a more direct way to support this sort of thing.  But until we do, we need to keep supporting the workaround that has long been available.

**Comment by DmitryArefiev** [comment]
@Tantacrul FYI

**Comment by Jojo-Schmitz** [comment]
Hmm, yeah, that case does require a workaround anyway, so can as well do with a few more extra steps I guess.

**Comment by MarcSabatella** [comment]
Good point.  At least, once the stacking order buttons are fixed, this will become possible.

**Comment by Tantacrul** [comment]
Since this is an existing issue with MS3, I think we should bump it to 4.x

There's a larger discussion about permitting these kinds of clever tricks without needing to figure out quite complex workarounds.... but I think it should be tackled with feature improvements to slurs in future. A topic for another day!

**Comment by MarcSabatella** [comment]
This is merged, so existing scores like my Reunion demo broke as expected.  However, it turns out the suggested methods to fix this don't work.  The "to front" / "to back" stacking order buttons are still broken (see https://github.com/musescore/MuseScore/issues/27771), and worse, if you use the forwards/backwards buttons to get the desired result, the values don't save.  I'll file a separate issue for that.

**Comment by MarcSabatella** [comment]
Turns out there was already an issue for the inability to save z-order for ties - https://github.com/musescore/MuseScore/issues/31056.
